File: flutter_app/lib/widgets/priority_badge.dart
import 'package:flutter/material.dart';
import '../models/todo.dart';

class PriorityBadge extends StatelessWidget {
  const PriorityBadge({super.key, required this.priority});
  final TodoPriority priority;

  @override
  Widget build(BuildContext context) {
    final color = switch (priority) {
      TodoPriority.high => Colors.red,
      TodoPriority.medium => Colors.orange,
      TodoPriority.low => Colors.green,
    };
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
      decoration: BoxDecoration(
        color: color.withValues(alpha: 0.15),
        border: Border.all(color: color),
        borderRadius: BorderRadius.circular(12),
      ),
      child: Text(
        priority.label,
        style: TextStyle(color: color, fontSize: 11, fontWeight: FontWeight.bold),
      ),
    );
  }
}

File: ios_swiftui/TodoApp/Features/Todo/Components/PriorityBadge.swift
import SwiftUI

struct PriorityBadge: View {
    let priority: TodoPriority

    var body: some View {
        Text(priority.label)
            .font(.caption2)
            .fontWeight(.semibold)
            .foregroundStyle(.white)
            .padding(.horizontal, 6)
            .padding(.vertical, 2)
            .background(color)
            .clipShape(Capsule())
    }

    private var color: Color {
        switch priority {
        case .low: .green
        case .medium: .orange
        case .high: .red
        }
    }
}
